File: src/Application Specific/Forms And Controls/Jobs/JobSheets/EngineerVisit/FRMChangePriority.vb
Public Class FRMChangePriority

    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        If Combo.GetSelectedItemValue(cboJobType) = 0 Then
            ShowMessage("Please select a priority", MessageBoxButtons.OK, MessageBoxIcon.Information)
        Else
            If Me.Modal Then
                Me.Close()
            Else
                Me.Dispose()
            End If
        End If
    End Sub

    Private Sub FRMChangePriority_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Dim dt As DataTable = DB.Picklists.GetAll(Entity.Sys.Enums.PickListTypes.JOWPriority).Table
        Dim dr() As DataRow = dt.Select("Name<>'RC - Recall'")
        Dim newDt As DataTable = dt.Clone
        Dim nr As DataRow
        For Each r As DataRow In dr
            nr = newDt.NewRow
            nr("ManagerID") = r("ManagerID")
            nr("Name") = r("Name")
            newDt.Rows.Add(nr)
        Next
        Combo.SetUpCombo(cboJobType, newDt, "ManagerID", "Name", Entity.Sys.Enums.ComboValues.Please_Select)
    End Sub

End Class

File: apollo/apollo.fsm/src/Application Specific/Forms And Controls/Jobs/JobSheets/EngineerVisit/FRMChangePriority.cs
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Data;
using System.Windows.Forms;

namespace FSM
{
    public partial class FRMChangePriority
    {
        public FRMChangePriority()
        {
            InitializeComponent();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            if (Conversions.ToDouble(Combo.get_GetSelectedItemValue(cboJobType)) == 0)
            {
                App.ShowMessage("Please select a priority", MessageBoxButtons.OK, MessageBoxIcon.Information);
            }
            else if (Modal)
            {
                Close();
            }
            else
            {
                Dispose();
            }
        }

        private void FRMChangePriority_Load(object sender, EventArgs e)
        {
            var dt = App.DB.Picklists.GetAll(Entity.Sys.Enums.PickListTypes.JOWPriority).Table;
            var dr = dt.Select("Name<>'RC - Recall'");
            var newDt = dt.Clone();
            DataRow nr;
            foreach (DataRow r in dr)
            {
                nr = newDt.NewRow();
                nr["ManagerID"] = r["ManagerID"];
                nr["Name"] = r["Name"];
                newDt.Rows.Add(nr);
            }

            var argc = cboJobType;
            Combo.SetUpCombo(ref argc, newDt, "ManagerID", "Name", Entity.Sys.Enums.ComboValues.Please_Select);
        }
    }
}